src/main/java/edu/cornell/kfs/fp/dataaccess/impl/RecurringDisbursementVoucherSearchDaoJdbc.java

```diff
@@ -0,0 +1,91 @@
+
+    @Override
+    public Collection<String> findSavedDvIdsSpawnedByRecurringDvForCurrentAndPastFiscalPeriods(Date currentFisalPeriodEndDate) {
+        final Date currentFisalPeriodEndDateForSql= currentFisalPeriodEndDate;
+        Collection<DisbursementVoucherDocument> dvsFound = null;
+
+        return getJdbcTemplate().execute(new ConnectionCallback<Collection<String>>() {
+            public Collection<String> doInConnection(Connection con) throws SQLException {
+                PreparedStatement selectStatement = null;
+                ResultSet queryResultSet = null;
+                List<String> finalResults = new ArrayList<String>();
+
+                try {
+                    String selectStatementSql = getSavedDvsSpawnedByRecurringDvForCurrentAndPastFiscalPeriodsSelectSql(currentFisalPeriodEndDateForSql);
+                    LOG.info("SQL Statement to obtain saved DVs spawned by RCDV for current and past fiscal period follows.");
+                    LOG.info("*************************************");
+                    LOG.info(selectStatementSql);
+                    LOG.info("*************************************");
+                    selectStatement = con.prepareStatement(selectStatementSql);
+                    queryResultSet = selectStatement.executeQuery();
+
+                    while (queryResultSet.next()) {
+                        finalResults.add(queryResultSet.getString(1));
+                        LOG.info("Found DV DocID=" +queryResultSet.getString(1)+ "=  DVCheckDateSameAsDueDate=" +queryResultSet.getString(2)+ "=");
+                    }
+                    queryResultSet.close();
+
+                } finally {
+                    if (queryResultSet != null) {
+                        try {
+                            queryResultSet.close();
+                        } catch (SQLException e) {
+                            LOG.error("Could not close ResultSet");
+                        }
+                    }
+                    if (selectStatement != null) {
+                        try {
+                            selectStatement.close();
+                        } catch (SQLException e) {
+                            LOG.error("Could not close selection PreparedStatement");
+                        }
+                    }
+                }
+                return finalResults;
+            }
+        });
+    }
+
+    private String getSavedDvsSpawnedByRecurringDvForCurrentAndPastFiscalPeriodsSelectSql(Date currentFisalPeriodEndDate) {
+        SimpleDateFormat formatter = new SimpleDateFormat("dd-MMM-yy", Locale.US);
+        return new String("SELECT B.DV_DOC_NBR, B.DV_CHECK_DT FROM KFS.FP_DV_DOC_T A, KFS.FP_RCDV_DTL_T B, CYNERGY.KREW_DOC_HDR_T C WHERE A.FDOC_NBR = B.DV_DOC_NBR AND C.DOC_HDR_ID = B.DV_DOC_NBR AND C.DOC_HDR_STAT_CD = '") +KewApiConstants.ROUTE_HEADER_SAVED_CD+ ("' AND B.DV_CHECK_DT <= '" +formatter.format(currentFisalPeriodEndDate)+ "'");
```

Comment:
Just a quick note, I would suggest using StringBuilder instead of StringBuffer, unless you need the thread-safety or regex-processing-related features of the latter.